Content.Client/SubFloor/TrayScannerSystem.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections.Generic;
+using System.Linq;
+using Content.Shared.SubFloor;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Maths;
+using Robust.Shared.Utility;
+
+namespace Content.Client.SubFloor;
+
+public class TrayScannerSystem : SharedTrayScannerSystem
+{
+    [Dependency] private IEntityLookup _entityLookup = default!;
+    [Dependency] private SubFloorHideSystem _subfloorSystem = default!;
+    [Dependency] private SharedContainerSystem _containerSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
```

Comment:
Also, should probably subscribe to component remove events to re-hide any revealed entities. When the scanner is deleted. Currently they're not destructible, so it's not that important. But if they ever are (or are consumed in crafting or something), then that would need to be added.